src/Microsoft.AspNet.OData.Shared/Formatter/Serialization/ODataResourceSerializer.cs

```diff
@@ -671,8 +682,16 @@ public virtual string CreateETag(ResourceContext resourceContext)
                     Name = complexProperty.Name
                 };
 
+                IList<ExpandedNavigationSelectItem> selectItems = null;
+                if (selectExpandNode.ExpandedNavigationPropertiesOnComplexTypes != null)
```

Comment:
>if (selectExpandNode.ExpandedNavigationPropertiesOnComplexTypes != null) [](start = 16, length = 72)

Let's discuss the followings with @mike

1. $select=Name&$expand=Np
2. $select=Name,NP&$expand=Np
3. $select=Name&$expand=Cp/Np
4. $select=Cp&$expand=Cp/Np
5. $select=Cp/Np&$expand=Cp/Np

Reply:
First 4 are supported but a path in select is not supported. 

---
In reply to: [274692314](https://github.com/OData/WebApi/pull/1738#discussion_r274692314) [](ancestors = 274692314)